lgc/interface/lgc/ElfNoteEntryInsertionUtil.h

```diff
@@ -0,0 +1,74 @@
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  ElfNoteEntryInsertionUtil.h
+ * @brief LLPC header file: declaration of lgc::AddNotesToELF interface
+ *
+ * @details The function AddNotesToELF adds given note entries to the given ELF.
+ ***********************************************************************************************************************
+ */
+
+#pragma once
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/CommandLine.h"
+
+namespace lgc {
+
+// Note entry that will be added to the note section.
+struct NoteEntry {
+  llvm::StringRef name;
+  llvm::ArrayRef<uint8_t> desc;
+};
+
+// =====================================================================================================================
+// Adds the given note entries to the given ELF if the given ELF has a note section.
+// Otherwise, it does nothing.
+//
+// ELF layouts before/after adding new "note" entries to the existing note section:
+//
+// |-------------|             |-------------|
+// | ELF header  |             | ELF header  |
+// |-------------|             |-------------|
+// | Sections    |             | Sections    |
+// | ...         |             | ...         |
+// |-------------|             |-------------|
+// | Note section|     ==>     | Note section|
+// | ...         |             |             |
+// |-------------|<--(Will be  | + New note  |
+// | ...         |    Shifted) |   entries   |
+// |-------------| |        |  | ...         |
+// | Section     | |        \->|-------------|
+// | headers     | V           | ...         |
+// | ...         |             |-------------|
+//                             | Section     |
+//                             | headers     |
+//                             | ...         |
+//
+void AddNotesToELF(llvm::SmallVectorImpl<char> &elf, llvm::ArrayRef<NoteEntry> notes, const char *noteSectionName);
```

Comment:
Naming niggles: In LGC, even a global func does not have an initial capital. And, unlike in LLVM, we treat an initialism as a word, so not all capitals. So the func should be "addNotesToElf".